### ISSUE TITLE
httplib2.Http() generates cache keys that exceed memcached max length

### ISSUE THREAD

**Issue by Alex3917** [httplib2/httplib2#1]
I'm creating an Http object with a cache like so:

httplib2.Http(cache=cache)

This fails when my API calls have long querystrings, because the max length of a memcached key is 250. As a proposed solution, maybe a hash of the querystring could be used as the cache key instead.

**Comment by rkhwaja** [comment]
This problem happens with the filesystem cache on windows too.

**Comment by galvez** [comment]
@rkhwaja I'm tagging this as an `enhancement` for now since it's an edge case -- will revisit when I'm done reviewing other open issues.

**Comment by 0Hughman0** [comment]
I think I'm running into this problem at the moment using the google API, as some of the query URLs are super long.

**Comment by Alex3917** [comment]
Yeah, the Google API is also where I ran into this. (Specifically doing searches using the Gmail API.)
